guides/CONTRIBUTING.md

```diff
@@ -18,26 +16,165 @@ When filing an issue, make sure to provide:
+
+#### Changes in native code 
+
+If your change is only related to the native code (for example you're fixing a bug), you **do not need** to create a PR in the `react-navigation` repository. `react-native-screens` is a separate library installed alongside `react-navigation`, so you can just increase version of `react-native-screens` in your package.json file.
+
+#### Changes in JS code or both
+
+If you're changing native-stack v5 (`src/native-stack`) you **do not need** to create a PR in the `react-navigation` repository.
+
+If you're changing native-stack v6, v7 you **need to** create a PR in the `react-navigation` repository, since the code belongs there now.
 
-### Pull requests
+If you're changing core functionality:
+- If it's a bug fix and does not change the interface, then you **do not need** to create a PR in `react-navigation` repository.
+- If you're adding a new feature or changing API, you **need to** create a PR in `react-navigation` repository.
 
-Pull requests are more than welcome 🚀
+> [!TIP]
+> As the rule of thumb, if you're changing the public interface, you need to prepare PRs for the `react-navigation` repository.
 
-We've created a template to help you with the process of submitting a pull request to this project. Remember to change or add documentation as needed and provide a code snippet (preferably added to `TestsExample/` project) that the developers could test your change on. In the PR template we've added a checklist for all the places where the documentation should be updated.
+### What is the flow to integrate with react-navigation
 
-If it's your first pull request [this guide](https://github.com/MarcDiethelm/contributing/blob/master/README.md) might be helpful for you.
+When you propose the changes that require creating a PR to the `react-navigation` repository, please follow these steps:
 
-### Naming convensions
+1. Create a PR with `react-native-screens` changes
+2. Create a PR with changes for native-stack in `react-navigation`
+3. Wait for both to pass the review and have all checks passed
+4. Merge `react-native-screens` changes
+5. Merge `react-native-navigation` changes
+6. Upgrade the version of main branch index reference in `react-native-screens` (usually, git checkout on a submodule should be sufficient)
+7. Post and merge ASAP upgrade of the main branch reference of `react-navigation`
 
-For commits and pull request names we follow a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+> [!WARNING]
+> Those steps are crucial, if you change the API in react-native-screens and won't merge react-navigation changes the libraries may go out of sync and crash i.e. because of not existing property. On the other hand, if you don't perform step 6 and 7, test examples and showcase app may stop working in react-native-screens.
 
-For branch names we use `@username/description-separated-by-dashes` convention.
```

Comment:
I think we should have this line present somewhere, not entirely removed.

Reply:
I guess that applies only for screens team, not the "external" contributors, as they'll create forks anyway (?)

Reply:
Yes, I think you're right.